best-practices/tech_lead.md

```diff
@@ -0,0 +1,367 @@
+  - (this is one of the reasons why the TL should not take blocking tickets or large tickets -- the TL priority is to keep the other devs unblocked)
+- If you believe a dev took a ticket but might unknowingly be missing some context or understanding, reach out.
+
+### With PO or Designer
+- Exchange info with PO or Designer - ask questions to clarify work in tickets, or relative priorities, or whatever.  
+
+## Board Grooming
+This is a big part of being tech lead.  The board is our way of knowing the state of work, our way to determine devs who might benefit from help without realizing they need it (e.g. "Naomi took that ticket but I need to make sure she knows about XXX"), our way to ensure there is always prioritized work available when a developer is looking for a task.
+
+Do PRs need review?  A good project board helps indicate this;  a TL ensures PRs are reviewed promptly (often by doing the reviews).
+
+The board can also be a tool for the PO or a Manager:  can they tell what is in progress, what is complete, how issues relate to the Epics (more below) for the work cycle, etc.?
+
+Q: ___How often should the TL look at / groom the board?___
+
+A: As often as possible throughout the day. Absolute minimum at 2-3 times / day.  
+
+Q: ___What does grooming the board mean?___
+
+A: Keeping the board useful for the devs, the PO, management; ensuring PR aren't waiting review; avoiding any blocked flow:
+
+- Is the ready column populated with prioritized, actionable work? (more below)
+- PRs needing review or re-review (more below)
+- Are open PRs associated with issues, if desired?
+- Are there issues in the new column that need to be triaged?
+- Are tickets in the correct columns?
+  - Can any issues be closed / moved to the closed column?
+  - Are all in progress tickets properly assigned and in the right column?
+  - Are things requiring PO review identified and communicated to PO?
+  - Ditto for design review?
+- Does the board reflect the current state?
+- Check board for new tickets created by others, ticket order in columns, etc.
+
+### Ready Column
+- Must _always_ be kept populated
+  - Depth of at least one ticket per developer
+- Each ticket here must be actionable for the developers in the work cycle
+  - any open questions for the PO or the designer aren't blocking the work (if so, ticket goes in backlog) ... and are indicated in the ticket
+  - vital background information provided (or links to it)
+  - volume of text in ticket isn't so overwhelming that developer backs away slowly. (Adding structure so it's more skimmable can help)
+  - it's okay to add comments to the ticket like "talk to me before you start coding for this" or "this will require some familiarity with xxx"
+
+## Pull Request Review
+
+___Quick Review___: It is up to the TL to ensure that PRs are reviewed quickly.  If the TL does a fair number of reviews, this also helps ensure new code is aligned with workcycle coding goals (see below).
```

Comment:
Pull request review should be the responsibility of the development team members, not the tech lead. Failure of anything in here is not the failure of the tech lead, but of the team members.

Reply:
"It is up to the TL to ensure that PRs are reviewed quickly." -- that doesn't mean the TL does all the reviews;  it means the TL is the person that keeps looking to make sure PRs are being reviewed promptly.  Which may or may not be followed by the TL doing any needed reviews.

Reply:
It's the difference between "responsibility" and "accountability" in the [RACI model](https://en.wikipedia.org/wiki/Responsibility_assignment_matrix#Key_responsibility_roles_in_RACI_model). In a nutshell, I think @ndushay is suggesting the whole team (including TL) is *responsible* for code review, and the TL (and only the TL) is *accountable* for code review writ large.

Reply:
Good point.  And maybe even "accountable" is a strong word, because it may imply that management would blame the TL if the team is not keeping up with reviews or the project is falling behind.  But it is certainly useful to have someone (e.g. the TL) notice that things are backed up, bring it to the team or PO at standup and possibly request certain people prioritize things.  Basically, project management type stuff as Naomi stated.